packages/studio-base/src/components/TimeBasedChart/downsample.ts

```diff
@@ -129,22 +129,25 @@ export function continueDownsample(
     // create a new interval when encountering a new label to preserve the transition from one label to another
     if (intFirst?.xPixel !== x || (intLast?.label != undefined && intLast.label !== datum.label)) {
       // add the min value from previous interval if it doesn't match the first or last of that interval
+      const newPoints: number[] = [];
       if (intMin && intMin.yPixel !== intFirst?.yPixel && intMin.yPixel !== intLast?.yPixel) {
-        indices.push(intMin.index);
+        newPoints.push(intMin.index);
       }
 
       // add the max value from previous interval if it doesn't match the first or last of that interval
       if (intMax && intMax.yPixel !== intFirst?.yPixel && intMax.yPixel !== intLast?.yPixel) {
-        indices.push(intMax.index);
+        newPoints.push(intMax.index);
       }
 
       // add the last value if it doesn't match the first
       if (intLast && intFirst?.yPixel !== intLast.yPixel) {
-        indices.push(intLast.index);
+        newPoints.push(intLast.index);
       }
 
       // always add the first datum of an new interval
-      indices.push(index);
+      newPoints.push(index);
+
+      indices.push(...newPoints.sort((a, b) => a - b));
```

Comment:
```suggestion
      indices.push(...newPoints.sort();
```
sad we can't do an in-place sort on a range of an array, but I can't figure out an obvious way without using a JS sort implementation, which might be slower.

Reply:
at any rate, N is strictly <= 4 here so it could be worse?

Reply:
if N is strictly <= 4, it seems likely that an in-place simple (insertion?) sort on the end of `indices` would be faster than creating a new array, sorting it, then pushing it onto `indices`.

Reply:
I agree with James here - this looks somewhat "wasteful" to the reader without an additional comment - either way a comment would be useful here on what behavior matters

Reply:
this could be clean actually if every `push` onto `indices` was a done with a `pushAndBubbleDown` function. you'd be doing an online insertion sort into `indices`.